### PR TITLE
Fix/wds color bgaccent

### LINKED ISSUE #24033: bgAccentHover and bgAccentActive (dark mode)

### PR DESCRIPTION
**Pull request by ichik**
## Description

tl;dr Better logic and looks for dark mode `bgAccent*`, using correct reference lightness in light mode 

Fixes #24033 #24209   

## Type of change
- Bug fix (non-breaking change which fixes an issue)
- New feature (non-breaking change which adds functionality)
- Chore (housekeeping or task changes that don't impact user perception)

## How Has This Been Tested?

- Manual

### Test Plan
Initial POC refinement, no testing necessary

### Dev activity
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] PR is being merged under a feature flag

### QA activity:
- [ ] Test plan has been approved by relevant developers
- [ ] Test plan has been peer reviewed by QA
- [ ] Cypress test cases have been added and approved by either SDET or manual QA
- [ ] Organized project review call with relevant stakeholders after Round 1/2 of QA
- [ ] Added Test Plan Approved label after reveiwing all Cypress test

### CODE REVIEW

File: app/client/packages/design-system/theming/src/color/DarkModeTheme.ts [hunk 1]
```diff
@@ -87,11 +90,75 @@ export class DarkModeTheme implements ColorModeTheme {
...
+      this.seedIsCold
+    ) {
+      color.oklch.l = this.bgAccent.oklch.l + 0.04;
+      color.oklch.c = this.bgAccent.oklch.c + 0.05;
+    }
+
+    if (
+      this.seedLightness >= 0.77 &&
+      this.seedLightness < 0.85 &&
+      !this.seedIsAchromatic &&
```
Comment (KelvinOm) [inline app/client/packages/design-system/theming/src/color/DarkModeTheme.ts:120]: We don't check for seedIsAchromatic at `lightMode`, are we sure it's not necessary there?

File: app/client/packages/design-system/theming/src/color/DarkModeTheme.ts [hunk 2]
```diff
@@ -87,11 +90,75 @@ export class DarkModeTheme implements ColorModeTheme {
...
+
+    if (
+      this.seedLightness >= 0.77 &&
+      this.seedLightness < 0.85 &&
+      this.seedIsAchromatic
+    ) {
+      color.oklch.l = this.bgAccent.oklch.l + 0.04;
+    }
+
+    if (this.seedLightness >= 0.85) {
```
Comment (KelvinOm) [inline app/client/packages/design-system/theming/src/color/DarkModeTheme.ts:135]: Also at `LightMode` we have checks `seedIsVeryLight` and `seedIsYellow`, maybe there should be something opposite?

**Review by KelvinOm — APPROVED** [review]

File: app/client/packages/design-system/theming/src/color/DarkModeTheme.ts (hunk 2 above)
Comment (ichik) [reply app/client/packages/design-system/theming/src/color/DarkModeTheme.ts:135]: That first condition `this.seedLightness < 0.3` is the `isVeryDark` check actually. I used this syntax since it's a bit more obvious then in the code that it's a “breakpoint” kind of logic here, so perhaps it's the light mode code that needs to be updated.

The check for `isYellow` in the this analogous range is not necessary: chromas are typically skewed towards light side, it's especially pronounced in Yellow hence the need for compensation there.
image
image
image
image
